src/SignalR/server/Core/src/HubConnectionContext.cs

```diff
@@ -313,13 +333,19 @@ private async Task TryWritePingSlowAsync()
         {
             try
             {
+                if (_connectionAborted)
```

Comment:
Just looking at this for the first time, I find it very funny that we have TryWritePingSlowAsync for the normal case when we actually write a ping, but save on the state machine allocation in the TryWritePingAsync failure case when it can't immediately acquire the write lock.

I think we might have officially jumped the shark with this fast/slow/awaited pattern 🤣

Reply:
![fonzie jumping the shark](https://www.bing.com/th?id=OGC.f19df97352ab27d690822061edae5113&pid=1.7&rurl=http%3a%2f%2fthisevilempire.com%2fblog%2fwp-content%2fuploads%2f2012%2f06%2ffonzie3.gif&ehk=HGrtug5%2byCIrp360lahQAw)